plugins/package-managers/cocoapods/src/main/kotlin/Podspec.kt

```diff
@@ -56,22 +53,22 @@ internal data class Podspec(
     }
 }
 
-internal fun String.parsePodspec(): Podspec = jsonMapper.readValue<Podspec>(this)
+private val JSON = Json { ignoreUnknownKeys = true }
+
+internal fun String.parsePodspec(): Podspec = JSON.decodeFromString<Podspec>(this)
 
 /**
  * Handle deserialization of the following two possible representations:
  *
  * 1. https://github.com/CocoaPods/Specs/blob/f75c24e7e9df1dac6ffa410a6fb30f01e026d4d6/Specs/8/5/e/SocketIOKit/2.0.1/SocketIOKit.podspec.json#L6-L9
  * 2. https://github.com/CocoaPods/Specs/blob/f75c24e7e9df1dac6ffa410a6fb30f01e026d4d6/Specs/8/5/e/FirebaseObjects/0.0.1/FirebaseObjects.podspec.json#L6
  */
-private class LicenseDeserializer : StdDeserializer<String>(String::class.java) {
-    override fun deserialize(parser: JsonParser, context: DeserializationContext): String {
-        val node = parser.codec.readTree<JsonNode>(parser)
-
-        return if (node.isTextual) {
-            node.textValue()
+private object LicenseSerializer : JsonTransformingSerializer<String>(String.serializer()) {
+    override fun transformDeserialize(element: JsonElement): JsonElement =
+        if (element is JsonObject) {
+            val type: String = (element["type"] as JsonPrimitive).content
```

Comment:
The explicit type should be dropped, and the `jsonPrimitive` extension function should be used (which requires explicit handling of the `null` case for `element["type"]`).

Reply:
Can you please provide a rationale for the latter? (I believe it is fine to assume that type is not null here, which this code does.)

Reply:
Explicit handling of the `null` case for `element["type"]` is a side effect of using `jsonPrimitive`, not the primary goal. But I think it's a good side effect as the original code also handled the `null` case explicitly via `textValueOrEmpty()`. If you want to document that you expect the field to exist in any case (and throw an exception otherwise), then you should use `getValue()` instead of `[]`.

Reply:
Note: I kept the explicit type for readability.